rllib/algorithms/algorithm_config.py

```diff
@@ -3648,12 +3664,30 @@ def _validate_evaluation_settings(self):
             )
 
         # If async evaluation is enabled, custom_eval_functions are not allowed.
-        if self.enable_async_evaluation and self.custom_evaluation_function:
+        if (
+            self.enable_async_evaluation
+            and self.custom_evaluation_function
+            and not self.uses_new_env_runners
+        ):
             raise ValueError(
                 "`config.custom_evaluation_function` not supported in combination "
                 "with `enable_async_evaluation=True` config setting!"
             )
 
+        # Customized asynchronous evaluation is only possible with the `EnvRunner API`.
+        if self.custom_async_evaluation_function and not self.uses_new_env_runners:
+            raise ValueError(
+                "`config.custom_async_evaluation_function` not supported in "
+                "combination with `RolloutWorker`s! Use `env_runner_cls="
+                "SingleAgentEnvrunner|MultiAgentEnvRunner`."
+            )
+        elif self.custom_async_evaluation_function and self.uses_new_env_runners:
```

Comment:
Can we fix these two if statements? One main if that checks for `self.custom_async_evaluation_function is not None`.
Then an if-else block within that checking whether we are on old RolloutWorkers (-> error) or on new EnvRunners (-> validate signature, etc..)